app/graphql/resolvers/add_asset_to_submission_resolver.rb

```diff
@@ -1,25 +1,36 @@
 # frozen_string_literal: true
 
 class AddAssetToSubmissionResolver < BaseResolver
-  def valid?
-    true
-  end
+  include Resolvers::Submissionable
 
   def run
-    submission = Submission.find_by(id: @arguments[:submission_id])
-    unless submission
-      raise(GraphQL::ExecutionError, 'Submission from ID Not Found')
-    end
+    check_submission_presence!
 
     unless matching_user(submission, @arguments&.[](:session_id)) || admin?
       raise(GraphQL::ExecutionError, 'Submission Not Found')
     end
 
     @arguments[:asset_type] ||= 'image'
 
-    asset = submission.assets.create!(@arguments.except(:session_id))
+    asset = submission.assets.create!(asset_params)
     SubmissionService.notify_user(submission.id) if submission.submitted?
 
     { asset: asset }
   end
+
+  private
+
+  # overwrites Resolvers::Submissionable
```

Comment:
if it's not clear what's going on here - please see docs in Resolvers::Submissionable. If still not clear - let's discuss how to implement it better